blueocean-git-pipeline/src/main/java/io/jenkins/blueocean/blueocean_git_pipeline/GitReadSaveService.java

```diff
@@ -0,0 +1,286 @@
+import org.kohsuke.stapler.StaplerRequest;
+
+import hudson.EnvVars;
+import hudson.Extension;
+import hudson.model.Item;
+import hudson.model.TaskListener;
+import hudson.plugins.git.GitException;
+import hudson.plugins.git.GitTool;
+import hudson.plugins.git.extensions.GitSCMExtension;
+import hudson.remoting.Base64;
+import hudson.util.LogTaskListener;
+import io.jenkins.blueocean.rest.impl.pipeline.ScmContentProvider;
+import jenkins.branch.MultiBranchProject;
+import jenkins.model.Jenkins;
+import jenkins.plugins.git.GitSCMSource;
+import jenkins.scm.api.SCMSource;
+import net.sf.json.JSONObject;
+
+@Extension
+public class GitReadSaveService extends ScmContentProvider {
+
+    @Override
+    public String getScmId() {
+        return "git";
+    }
+
+    @Override
+    public String getApiUrl(Item item) {
+        return null;
+    }
+    
+    public static class GitReadSaveRequest {
+        final String readUrl;
+        final String writeUrl;
+        final String readCredentialId;
+        final String writeCredentialId;
+        final String branch;
+        final String commitMessage;
+        final String sourceBranch;
+        final String filePath;
+        final byte[] contents;
+        final GitClient git;
+        final GitTool gitTool;
+        final File repositoryPath;
+        final GitSCMSource gitSource;
+
+        public GitReadSaveRequest(
+                String readUrl, String writeUrl,
+                String writeCredentialId, String readCredentialId,
+                String branch, String commitMessage,
+                String sourceBranch, String filePath,
+                byte[] contents, GitTool gitTool, GitSCMSource gitSource)
+                throws IOException, InterruptedException {
+            this.readUrl = readUrl;
+            this.writeUrl = writeUrl;
+            this.readCredentialId = readCredentialId;
+            this.writeCredentialId = writeCredentialId;
+            this.branch = branch;
+            this.commitMessage = commitMessage;
+            this.sourceBranch = sourceBranch;
+            this.filePath = filePath;
+            this.contents = contents;
+            this.gitTool = gitTool;
+            this.gitSource = gitSource;
+            this.repositoryPath = Files.createTempDirectory("git").toFile();
+
+            EnvVars environment = new EnvVars();
+            TaskListener taskListener = new LogTaskListener(Logger.getAnonymousLogger(), Level.ALL);
+            String gitExe = gitTool.getGitExe();
+            git = Git.with(taskListener, environment)
+                    .in(repositoryPath)
+                    .using(gitExe)
+                    .getClient();
+        }
+        
+        byte[] read() throws IOException, InterruptedException {
+            try {
+                // thank you test for how to use something...
+                // https://github.com/jenkinsci/git-client-plugin/blob/master/src/test/java/org/jenkinsci/plugins/gitclient/GitClientTest.java#L1108
```

Comment:
Remove comments to example code that someone might need to use in the future? I'd rather not.